Cython/Compiler/Parsing.py

```diff
@@ -3077,18 +3087,49 @@ def p_with_gil(s):
     else:
         return 0
 
-def p_exception_value_clause(s):
+def p_exception_value_clause(s, ctx):
+    """
+    Parse exception value clause.
+
+    Maps clauses to exc_check / exc_value / exc_clause as follows:
+     ______________________________________________________________________
+    |                             |             |             |            |
+    | Clause                      | exc_check   | exc_value   | exc_clause |
+    | ___________________________ | ___________ | ___________ | __________ |
+    |                             |             |             |            |
+    | <nothing> (default func.)   | True        | None        | False      |
+    | <nothing> (cdef extern)     | False       | None        | False      |
+    | noexcept                    | False       | None        | True       |
+    | except <val>                | False       | <val>       | True       |
+    | except? <val>               | True        | <val>       | True       |
```

Comment:
Readding [comments](https://github.com/cython/cython/pull/4670#discussion_r955128309) that got prematurely outdated:
> Just realized that `except *` is a thing as well. It's somewhat confusing that this isn't something like `except? *` (as otherwise, `except?` means `check=True` and `except` means `check=False`), but it's black-on-white in the [docs](https://cython.readthedocs.io/en/latest/src/tutorial/pure.html#static-typing).

```suggestion
    | except <val>                | False       | <val>       | True       |
    | except? <val>               | True        | <val>       | True       |
    | except *                    | True        | None        | True       |
```
and
> > as otherwise, `except?` means `check=True` and `except` means `check=False`
> 
> That horse has long left the barn, but it would be more intuitive to me (aside from jibing with `except *`) if `except` were to check for exceptions and `except?` didn't ("?" -> more uncertainty...).